src/middleware/validator/courses.validator.ts

```diff
@@ -0,0 +1,12 @@
+import { check } from 'express-validator'
+
+import validate from './generic.validator'
+
+const name = check('name').isString().trim().isLength({ max: 128 })
+const semester = check('semester').isString().trim().isLength({ max: 128 })
+const number = check('number').isString().trim().isLength({ max: 128 })
+// Should we validate dates?
```

Comment:
Yes. At the very least they should be on the request, and they should be dates. We may want to add more, like limiting creating assignments that are due in the past or something; but tbh, that's probably overly limiting. At the very least we'll want to validate that the date fields are on the request and that they look like date time strings.